src/Uno.UI/WinRT/Extensions/UI/Popups/MessageDialogContentDialog.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Uno.UI.Helpers.WinUI;
+using Windows.System;
+using Windows.UI.Popups;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Input;
+
+namespace Uno.UI.WinRT.Extensions.UI.Popups;
+
+internal partial class MessageDialogContentDialog : ContentDialog
+{
+	private readonly MessageDialog _messageDialog;
+	private readonly List<IUICommand> _commands;
+	private readonly uint _cancelCommandIndex;
+
+	public MessageDialogContentDialog(MessageDialog messageDialog)
+	{
+		DefaultStyleKey = typeof(ContentDialog);
+		_messageDialog = messageDialog ?? throw new ArgumentNullException(nameof(messageDialog));
+
+		// WinUI provides a modern style for ContentDialog, which is not applied automatically - force it
+		if (Application.Current.Resources.TryGetValue("DefaultContentDialogStyle", out var resource) && resource is Style winUIStyle)
+		{
+			Style = winUIStyle;
+		}
+
+		_commands = _messageDialog.Commands.ToList();
+		_cancelCommandIndex = _messageDialog.CancelCommandIndex;
+
+		Content = _messageDialog.Content;
+		Title = _messageDialog.Title;
+
+		if (_commands.Count == 0)
+		{
+			// Only show a close button
+			var closeText = ResourceAccessor.GetLocalizedStringResource("NavigationCloseButtonName");
+			_commands.Add(new UICommand(closeText));
+		}
+
+		PrimaryButtonText = _commands[0].Label;
+		SecondaryButtonText = _commands.Count > 1 ? _commands[1].Label : null;
+		CloseButtonText = _commands.Count > 2 ? _commands[2].Label : null;
```

Comment:
Although it would make more logical sense, the end result will be the same (the `ContentDialog` will just display primary button in the location where close button would be). I would actually prefer if they didn't explicitly "name" those buttons, but what can you do :-D .